### ISSUE TITLE
Sort by multiple criteria (Call for testing)

### ISSUE THREAD

**Issue by rzelnik** [ransome1/sleek#135]
It would be great to have two (or more) sorting criteria, for example Sort by Due Date and then by Priority.

**Comment by nutnutwin** [comment]
Agreed, Project and Due date will be so great

**Comment by ransome1** [comment]
@rzelnik , while looking into how to approach this feature, I was rereading your post @rzelnik. If sorting/grouping by due date first and then by priority is what you want to do, it should be possible already. If you change the sorting method to due date, the next sorting method will be priority. That only as a hint.

The real feature of course is to make every possible combination defineable. Let#s see what can be done ;)

**Comment by rzelnik** [comment]
Yes, I already noticed that when I sort by Due date, priority is the (hidden) second sorting option. The another drop-down menu for the secondary sorting option would be useful - it would be it clearer and easier to understand and it would add more customizability. However, one visible sorting option and one hidden/automatic is also good.

**Comment by ransome1** [comment]
@rzelnik , @nutnutwin , I added a first draft of this function to the latest pre release: https://github.com/ransome1/sleek/releases/tag/v1.0.7

At this point I could really need your support. If you can spare a bit of time please download it and give it a proper testing. I'm curious if this works well in your environment.

**Comment by zerodat** [comment]
Hi, @ransome1.  I built my own AppImage for 1.0.7 and the multiple sort criteria feature seems to work for me.  Very cool!  However, I did notice one thing.  If I drag on the "==" beside the criterion, it moves by itself without the criterion and leaves the criterion on a separate line.

**Comment by 9z0** [comment]
Why didn't it occur to me that you are using Sleek to task manage Sleek development?! :D

A few notes/possible improvements:
- Drag and drop seems alright. You can also use little up/down arrows that appear on hovering.
- Perhaps an option to sort by the todo text itself.
- Another useful thing is sort "as in file" = disable any sorting. It can be a toggle next to the heading "Sort by".
- A toggle to exclude an option from the sorting entirely. For example, don't sort by context, not even last. If you want to continue the drag-and-drop method, you can add a line that if you drag something below it, it gets disabled (gray).
```
                   [ x ] Due date  ↕
                   [ x ] Priority  ↕
Sort by [ x ]      [ x ] Projects  ↕
                   [ x ] Todo text ↕
                   ----------------
                   [   ] Contexts  ↕
```

- Maybe also quick sorting toggles (not permanent, for the current view and clear when you clear filters)

**Comment by ransome1** [comment]
Thanks for your feedback so far. If you're wondering, why the filter buttons do not work any longer; Please redownload the release ;)

**Comment by ransome1** [comment]
@9z0 , @nutnutwin . I replaced the drag and drop implementation with one that is a bit more joyable ;)

@9z0 , I appreciate your suggestions, but I don't think I see the big benefits for the majority of sleeks users. 

sleeks approach in general is to keep the GUI as clean as possible. I rather make sleek a simple experience for 90% of the users than an overly functional one for 10%, that don't benfit or even confuse the 90%. But of course I understand that this is only my opinion and there will be plenty of users out there who disagree.

**Comment by 9z0** [comment]
No problem, although I think you underestimate the average sleek user!

However, I'd still recommend "sort by the todo text itself".

**Comment by ransome1** [comment]
@9z0 ah ok, now I understand what you are referring to in the other ticket.

**Comment by 9z0** [comment]
@ransome1 I will respond [here](#151) if it is OK with you.

**Comment by ransome1** [comment]
@rzelnik this feature has been released with 1.0.7: https://github.com/ransome1/sleek/releases/tag/v1.0.7

I am going to close here. If there are any issues with this feature, please report back here and we will reopen.

**Comment by 9z0** [comment]
I can also confirm that the setting in `user-preferences.json` works.

I removed the sort criteria that I don't want and sleek used the remaining ones and then in-file sorting as expected. 👍
